src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetriesBehavior.cs

```diff
@@ -28,28 +29,26 @@ public FirstLevelRetriesBehavior(FlrStatusStorage storage, FirstLevelRetryPolicy
                 var messageId = context.Message.MessageId;
                 var pipelineUniqueMessageId = messageId;
 
-                var numberOfFailures = storage.GetFailuresForMessage(pipelineUniqueMessageId);
+                storage.MarkForFirstLevelRetry(pipelineUniqueMessageId, ExceptionDispatchInfo.Capture(ex));
```

Comment:
Good catch! This was a bug - if FLR was disabled than the header passed to custom SLR policy would have a wrong value:
 * I've changed the name of `FailureInfo` property to reflect the intent - now it's `FLRetries`
 * I've added test to cover this scenario
 * I've moved incrementing the counter to after the notification call